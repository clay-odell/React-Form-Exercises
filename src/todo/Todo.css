.todo-item {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    border: 1px solid #ccc;
    margin-bottom: 10px;
    border-radius: 5px;
}

.todo-text {
    margin: 0;
    flex: 1;
}

.remove-button {
    background: red;
    color: white;
    border: none;
    border-radius: 50%;
    width: 20px;
    height: 20px;
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-left: 10px; 
}
