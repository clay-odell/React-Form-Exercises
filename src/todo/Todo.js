import React from 'react';
import './Todo.css';
const Todo = ({ todo, removeTodo }) => {
  return (
    <div className="todo-item">
      <p className="todo-text">{todo.text}</p>
      <button className="remove-button" onClick={() => removeTodo(todo.id)}>X</button>
    </div>
  );
};

export default Todo;
