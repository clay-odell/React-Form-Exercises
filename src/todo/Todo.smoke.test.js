import React from 'react';
import { render } from '@testing-library/react';
import Todo from './Todo.js';

const mockTodo = {
  text: 'Test Todo',
  id: 1
};

const mockRemoveTodo = jest.fn();

it("renders without crashing", () => {
    render(<Todo todo={mockTodo} removeTodo={mockRemoveTodo} />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<Todo todo={mockTodo} removeTodo={mockRemoveTodo} />);
    expect(asFragment()).toMatchSnapshot();
});
