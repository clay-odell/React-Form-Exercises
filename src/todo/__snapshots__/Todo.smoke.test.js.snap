// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-item"
  >
    <p
      class="todo-text"
    >
      Test Todo
    </p>
    <button
      class="remove-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
